feature/security/gnsi/acctz/tests/record_subscribe_partial/record_subscribe_partial_test.go

```diff
@@ -0,0 +1,864 @@
+		rpcPayload:           "",
+		succeeded:            false,
+		expectedStatus:       acctz.SessionInfo_SESSION_STATUS_OPERATION,
+		expectedAuthenType:   acctz.AuthnDetail_AUTHN_TYPE_UNSPECIFIED,
+		expectedAuthenStatus: acctz.AuthnDetail_AUTHN_STATUS_UNSPECIFIED,
+		expectedIdentity:     failUsername,
+	})
+
+	ctx = context.Background()
+	ctx = metadata.AppendToOutgoingContext(ctx, "username", successUsername)
+	ctx = metadata.AppendToOutgoingContext(ctx, "password", successPassword)
+
+	req := &p4pb.CapabilitiesRequest{}
+
+	payload, err := anypb.New(req)
+	if err != nil {
+		t.Fatal("failed creating anypb payload")
+	}
+
+	startTime = time.Now()
+
+	_, err = p4rtclient.Capabilities(ctx, req)
+	if err != nil {
+		t.Fatalf("error fetching p4rt capabilities, error: %s", err)
+	}
+
+	// remote from the perspective of the router
+	remoteAddr, err := net.ResolveTCPAddr("tcp", addrObj.String())
+	if err != nil {
+		t.Fatalf("failed resolving p4rt remote addr, error: %s", err)
+	}
+	localAddr, err := net.ResolveTCPAddr("tcp", target)
+	if err != nil {
+		t.Fatalf("failed resolving p4rt local addr, error: %s", err)
+	}
+
+	records = append(records, rpcRecord{
+		startTime:            startTime,
+		doneTime:             time.Now(),
+		rpcType:              acctz.GrpcService_GRPC_SERVICE_TYPE_P4RT,
+		rpcPath:              "/p4.v1.P4Runtime/Capabilities",
+		rpcPayload:           payload.String(),
+		localIp:              localAddr.IP.String(),
+		localPort:            uint32(localAddr.Port),
+		remoteIp:             remoteAddr.IP.String(),
+		remotePort:           uint32(remoteAddr.Port),
+		succeeded:            true,
+		expectedStatus:       acctz.SessionInfo_SESSION_STATUS_OPERATION,
+		expectedAuthenType:   acctz.AuthnDetail_AUTHN_TYPE_UNSPECIFIED,
+		expectedAuthenStatus: acctz.AuthnDetail_AUTHN_STATUS_SUCCESS,
+		expectedAuthenCause:  "authentication_method: local",
+		expectedIdentity:     successUsername,
+	})
+
+	return records
+}
+
+func getServiceTarget(t *testing.T, dut *ondatra.DUTDevice, service introspect.Service) string {
+	// this shouldn't happen really, but fallback to dut name for target addr
+	defaultAddr := dut.Name()
+
+	var defaultPort uint32
+
+	switch service {
+	case introspect.GNMI:
+		defaultPort = 9339
+	case introspect.GNOI:
+		defaultPort = 9339
+	case introspect.GRIBI:
+		defaultPort = 9340
+	case introspect.P4RT:
+		defaultPort = 9559
+	}
+
+	target := introspect.DUTDialer(t, dut, service).DialTarget
+	_, err := net.ResolveTCPAddr("tcp", target)
+	if err != nil {
+		t.Logf("failed resolving %s target %s, will use default values", service, target)
+		target = fmt.Sprintf("%s:%d", defaultAddr, defaultPort)
+	}
+	t.Logf("Target for %s service: %s", service, target)
+	return target
+}
+
+func prettyPrint(i interface{}) string {
+	s, _ := json.MarshalIndent(i, "", "\t")
+	return string(s)
+}
+
+func TestAccountzRecordSubscribePartial(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
```

Comment:
a bunch of this setup is also duplicative, and we would be better serviced by creating a common library to use here and the other accountz tests.

Reply:
Same as previous comment

Reply:
thnx!